src/Wrapper.js

```diff
@@ -311,6 +333,102 @@ class Wrapper extends React.PureComponent {
   }
 }
 
+const SwipeContainer = (() => {
+  let _progress = 0
+
+  const SwipeContainer = ({
+    children,
+    enabled,
+    menuPanelOpened,
+    onMenuPanelClose,
+    onMenuPanelOpen,
+    width,
+  }) => {
+    const oneThirdWindowWidth = width / 3
+
+    if (!enabled) {
+      return <Container>{children(Number(menuPanelOpened))}</Container>
+    }
+
+    return (
+      <Gesture passive={{ passive: true }}>
+        {({
+          delta: [xDelta, yDelta],
+          direction: [xDir, yDir],
+          down,
+          event,
+          initial: [xInitial],
+          xy: [x],
+        }) => {
+          if (!down && _progress > 0 && _progress < 1) {
+            _progress = !menuPanelOpened
+              ? _progress > THRESHOLD_PROGRESS
+                ? 1
+                : 0
+              : _progress < THRESHOLD_PROGRESS
+              ? 0
+              : 1
+            if (_progress === 1) {
+              setTimeout(() => {
+                _progress = 0
+                onMenuPanelOpen()
+              }, 1)
+            }
+            if (_progress === 0) {
+              setTimeout(() => {
+                onMenuPanelClose()
+              }, 1)
+            }
+            return <Container>{children(_progress)}</Container>
+          }
+
+          let progress = _progress || Number(menuPanelOpened)
+
+          if (
+            (progress > 0 && progress < 1) ||
+            (down &&
+              xDelta &&
```

Comment:
I’m not sure why this check (`xDelta`) and `xDir` are needed for?

Reply:
"Intent", I check this to make sure the user started a horizontal touch move.

Reply:
But is it really important to check if they are not `0`? Nothing should move if either `xDelta` or `xDir` is at `0`?

If it is needed, then it would be nice to make the tests explicit: `xDelta !== 0` and `xDir !== 0`.